### DISCUSSION TITLE
Can't get FilePond window object from a custom livewire component

### DISCUSSION THREAD

**Discussion by z4yed** [filamentphp/filament#14351] [Help]
### Package

filament/filament

### Package Version

v3.2

### Laravel Version

v11.9

### PHP Version

PHP 8.2

### Problem description

Can we access FilePond window object from custom component? Or do we need to manually install that npm package within filament? I think FilePond is already builtin into Filament!

### Expected behavior

FilePond object is available on forms that has FileUpload component. I wish we can access this in all all pages. 

### Steps to reproduce

Simple create a custom page component in filament. Navigate to that page and check the window object. 

Or clone the reproduction repository and:
1. `composer install`
2. `php artisan migrate:fresh --seed`
3. Login to `/admin` with `test@example.com` and `password`.
4. Navigate to UserResource and click `View Profile`. It is a custom component. 
5. Inspect and check window.FilePond object. It is `undefined`. 

### Reproduction repository (issue will be closed if this is not valid)

https://github.com/z4yed/filament-filepond-issue

### Donate 💰 to fund this issue

- You can donate funding to this issue. We receive the money once the issue is completed & confirmed by you.
- 100% of the funding will be distributed between the Filament core team to run all aspects of the project.
- Thank you in advance for helping us make maintenance sustainable!

<a href="https://polar.sh/filamentphp/filament/issues/14323">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/filamentphp/filament/issues/14323/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/filamentphp/filament/issues/14323/pledge.svg">
</picture>
</a>

**Comment by z4yed** [comment]
<img width="1439" alt="Screenshot 2024-09-23 at 12 30 07 PM" src="https://github.com/user-attachments/assets/97a6d610-7736-4e61-a8a7-f53b5d541bc4">

**Comment by sprtk-ches** [comment]
Unless I'm mistaken, `window.Filepond` is only instantiated if you have a FileUpload component present since it's loaded asynchronously. So you can't just access it anywhere.

https://github.com/filamentphp/filament/blob/fc08789512d32bba1e72a0b90bf35e93be8089d2/packages/forms/resources/views/components/file-upload.blade.php#L32

**Comment by z4yed** [comment]
@sprtk-ches In that case, should we install that package separately within Filament using esbuild?

**Comment by sprtk-ches** [comment]
> @sprtk-ches In that case, should we install that package separately within Filament using esbuild?

I guess it really depends on what you're trying to do? Either way, this is more of a discussion than an issue

**Comment by zepfietje** [comment]
Closing this issue as it's not a bug.
